common/azure/src/main/kotlin/com/cosmotech/api/azure/adx/AzureDataExplorerClient.kt

```diff
@@ -230,19 +231,22 @@ class AzureDataExplorerClient(
       scenarioRunWorkflowEndTime: T,
   ): Boolean {
     val databaseName = getDatabaseName(organizationId, workspaceKey)
-    val failureQueryPrimaryResults =
-        this.kustoClient.execute(
-                databaseName,
-                """
+    val failureQueryResultsJsonString =
+        this.kustoClient.executeToJsonResult(
+            databaseName,
+            """
                 .show ingestion failures
                 | where  Table  == 'ProbesMeasures'
                 | order by FailedOn desc
                 | project FailedOn
             """,
-                ClientRequestProperties().apply {
-                  timeoutInMilliSec = TimeUnit.SECONDS.toMillis(REQUEST_TIMEOUT_SECONDS)
-                })
-            .primaryResults
+            ClientRequestProperties().apply {
+              timeoutInMilliSec = TimeUnit.SECONDS.toMillis(REQUEST_TIMEOUT_SECONDS)
+            })
+
+    val failureQueryPrimaryResults =
+        KustoOperationResult(failureQueryResultsJsonString, "v1").primaryResults
```

Comment:
Could be interesting to extract the version "v1" here and explain with you use v1 instead of v2.
As far as I know, we do not know if the version "v1" will be supported/used for a long time, so it could be great to use a global constant instead.
Maybe add a link to the implementation for information:
https://github.com/Azure/azure-kusto-java/blob/03aabae864c68bca07519c541a02ffaef4f85c00/data/src/main/java/com/microsoft/azure/kusto/data/KustoOperationResult.java